rest/user_agent_version.go

```diff
@@ -39,7 +39,7 @@ func (uav *UserAgentVersion) parse() {
 	// Compile regex
 	re1, err := regexp.Compile(regex)
```

Comment:
Unrelated to redaction work, but it caught my eye. I'm happy to file an issue if you agree.

It would be nice to hoist this up into a package level variable to pre-compile it and avoid the overhead of compiling on each call to `parse()`, and will give a bit more visibility of broken regexes (i.e. Panic on SGW startup):

```go
userAgentRegex := regexp.MustCompile(`CouchbaseLite/([0-9]*)\.([0-9]*).*`)
```

Reply:
Yep good catch, I don't see any good reason to repeatedly recompile this regex since it's completely static.

Reply:
Raised #3384